src/ome.ts

```diff
@@ -0,0 +1,226 @@
+    let field = '0';
+
+    // TEMP: try to support plates with plate/acquisition/row/column hierarchy
+    // Where plate.acquisitions = [{'path': '0'}]
+    let acquisitionPath = plateAttrs?.acquisitions?.[0]?.path;
+    acquisitionPath = acquisitionPath ? acquisitionPath + '/' : '';
+
+    // imagePaths covers whole plate (not sparse) - but some will be '' if no Well
+    const imagePaths = rowNames.flatMap(row => {
+        return columnNames.map(col => {
+            let wellPath = `${acquisitionPath}${row}/${col}/`;
+            return wellPaths.includes(wellPath) ? `${wellPath}${field}/` : '';
+        });
+    });
+
+    // Find first valid Image, loading each Well in turn...
+    let imageAttrs = undefined;
+    async function getImageAttrs(path: string): Promise<any> {
+        if (path === '') return
+        try {
+            return await getJson(store, `${path}.zattrs`);
+        } catch (err) {
+        }
+    }
+    for (let i = 0; i < imagePaths.length; i++) {
+        imageAttrs = await getImageAttrs(imagePaths[i]);
+        if (imageAttrs) {
+            break;
+        }
+    }
+
+    // Lowest resolution is the 'path' of the last 'dataset' from the first multiscales
+    let resolution = imageAttrs.multiscales[0].datasets.slice(-1)[0].path;
+
+    // Create loader for every Well. Some loaders may be undefined if Wells are missing.
+    async function createLoaderFromPath(path: string): Promise<ZarrLoader | undefined> {
+        try {
+            const data = await openArray({ store, path: `${path}${resolution}/` });
+            let l = createLoader([data]);
+            return l;
+        } catch (err) {
+            console.log(`Missing Well at ${path}`);
+        }
+    }
+    const promises = await pMap(imagePaths, createLoaderFromPath, { concurrency: 10 });
+    let loaders = await Promise.all(promises);
+
+    const loader = loaders.find(Boolean);
+
+    // Load Image to use for channel names, rendering settings, sizeZ, sizeT etc.
+    const sourceData: SourceData = loadOME(config, imageAttrs.omero, loader as ZarrLoader);
+
+    sourceData.loaders = loaders;
+    sourceData.name = plateAttrs.name || "Plate";
+    sourceData.rows = rows;
+    sourceData.columns = columns;
+    // Us onClick from image config or Open Well in new window
+    sourceData.onClick = config.onClick || ((info: any) => {
+        let gridCoord = info.gridCoord;
+        if (!gridCoord) {
+            return;
+        }
+        const { row, column } = gridCoord;
+        let { source } = sourceData;
+        if (typeof source === 'string' && !isNaN(row) && !isNaN(column)) {
+            if (source.endsWith('/')) {
+                source = source.slice(0, -1);
+            }
+            let imgSource = `${source}/${acquisitionPath}${rowNames[row]}/${columnNames[column]}/`;
+            window.open(window.location.origin + '?source=' + imgSource);
+        }
+    })
+    return sourceData;
```

Comment:
It seems like the difference between this callback and the one defined within `loadOMEWell` is just `imgSource`. Probably hard to make a shared function because both rely heavily on being closures, though.